### DISCUSSION TITLE
💡 Question: Does this Pixi-graph support combo grouping of nodes for the data visualization?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [zakjan/pixi-graph#15] [Ideas]
Hi @zakjan,

> _Not sure if this is the correct to pose a question/giving feedback regarding pixi-graph._
> _Maybe enabling [Github Discussions tab](https://docs.github.com/en/discussions) for this repository might be a good 😄_

Firstly I want to say I'm a big fan of this library
- I like enjoy it's API & Typescript support
- I like its philosophy and that it leverages the [graphology library](https://github.com/graphology/graphology)
-  I found out about it via your Neo4j tech talk "[Building a Graph Visualization with PIXI.js](https://www.youtube.com/watch?v=pe2PgFVvtLs)"

With Pixigraph (_or maybe you know a pixi extension package_) is possible to render / group nodes in a "combo" style manner?
In the data visualization community, I haven't seen many libraries offer this feature. 

## Question(s)
- is this something that this library should handle or could this potentially be something worth discussing with the [Graphology](https://github.com/graphology/graphology) team?

The only 2 libraries that I am aware of that offer this functionality are:
- [Open Source] - Ant Visualization group's [G6 Typescript](https://g6.antv.vision/en/docs/manual/middle/elements/combos/defaultCombo) library ([Graphin](https://graphin.antv.vision/en-US), a react wrapper around G6) 
- [PAID license] - Cambridge Intelligence's [Keylines](https://cambridge-intelligence.com/keylines/) JS library & [Regraph](https://cambridge-intelligence.com/regraph/) (React wrapper around Keylines library)
- [Unmaintained since 2017-ish] -- [Cisco NextUI Node sets](https://youtu.be/gBsUDu8aucs?t=52)
   - [example](https://github.com/NeXt-UI/next-tutorials/blob/master/tutorials/tutorial-003-01.md)

![CleanShot 2022-01-10 at 19 20 40](https://user-images.githubusercontent.com/6743796/148875745-3030527a-5309-42b1-bf64-f470f031893c.gif)

https://user-images.githubusercontent.com/6743796/148876265-8cda468d-0d67-4afc-857e-667673b8676e.mp4

https://user-images.githubusercontent.com/6743796/148875939-d1ca9d20-7fcd-42e5-bf37-3a0a9f85fcc6.mp4

![image](https://user-images.githubusercontent.com/6743796/148875546-3ae5049e-39f3-4fb7-b9a8-5783560d7fd1.png)

## References
- [Overview of combos](https://g6.antv.vision/en/docs/manual/middle/elements/combos/defaultCombo)
- [Combo article by Cambridge intelligence 1](https://cambridge-intelligence.com/combos/)
- [Combo article by Cambridge intelligence 2](https://cambridge-intelligence.com/graph-visualization-rectangular-combos/)
- [Combo article by Cambridge Intelligence 3](https://cambridge-intelligence.com/category/keylines/combos/)

**Comment by zakjan** [comment]
Hi @cliffordfajardo, thanks for the interesting topic. Combos are indeed an interesting advanced visualization feature. As PIXI is a general-purpose drawing library, it would be possible to implement such feature, given enough resources. However, currently I'm occupied with other projects. I would recommend to raise the topic for discussion in [Sigma.js](https://github.com/jacomyal/sigma.js/), which is the primary visualization library using Graphology.

**Comment by cliffordfajardo** [comment]
Update in case any folks see this later G6 now supports the combo's I illustrated above
- https://github.com/antvis/G6/discussions/3435
